bin/docker-to-squash.py

```diff
@@ -0,0 +1,1430 @@
+          image_hash = tag_to_hash.get(image, None)
+          if image_hash:
+            logging.debug("image tag exists for %s", image)
+          else:
+            logging.info("Not removing %s. Image tag doesn't exist", image)
+            continue
+        manifest_path = hdfs_manifest_dir + "/" + image_hash
+        if does_hdfs_entry_exist(manifest_path, raise_on_error=False):
+          logging.debug("image manifest for %s exists: %s", image, manifest_path)
+        else:
+          logging.info("Not removing %s. Image manifest doesn't exist: %s", image, manifest_path)
+          continue
+
+        delete_list.append(manifest_path)
+
+        manifest, manifest_hash = get_hdfs_manifest_from_path(manifest_path)
+
+        config_hash = get_config_hash_from_manifest(manifest)
+        logging.debug("config_hash: %s", config_hash)
+
+        delete_list.append(hdfs_config_dir + "/" + config_hash)
+
+        layers = get_layer_hashes_from_manifest(manifest, False)
+        layers_paths = []
+        for layer in layers:
+          if layer not in layers_to_keep:
+            layers_paths.append(hdfs_layers_dir + "/" + layer + ".sqsh")
+        delete_list.extend(layers_paths)
+
+        logging.debug("delete_list: %s", delete_list)
+
+        remove_image_hash_from_dicts(hash_to_tags, tag_to_hash, image_hash)
+
+    write_local_image_tag_to_hash(local_image_tag_to_hash, hash_to_tags)
+    atomic_upload_mv_to_hdfs(local_image_tag_to_hash, image_tag_to_hash,
+                                     hdfs_root, replication,
+                                     image_tag_to_hash_hash)
+
+    hdfs_rm(delete_list)
+
+  finally:
+    if working_dir:
+      if os.path.isdir(working_dir):
+        shutil.rmtree(working_dir)
+
+def add_remove_tag(args):
+  pull_format = args.pull_format
+  hdfs_root = args.hdfs_root
+  image_tag_to_hash = args.image_tag_to_hash
+  replication = args.replication
+  sub_command = args.sub_command
+  images_and_tags = args.images_and_tags
+
+  hdfs_manifest_dir = hdfs_root + "/manifests"
+  working_dir = None
+
+  try:
+    working_dir = get_working_dir(args.working_dir)
+    local_image_tag_to_hash = os.path.join(working_dir, os.path.basename(image_tag_to_hash))
+    hash_to_tags, tag_to_hash, image_tag_to_hash_hash = populate_tag_dicts(hdfs_root,
+                                                                           image_tag_to_hash,
+                                                                           local_image_tag_to_hash)
+
+    for image_and_tag_arg in images_and_tags:
+      if sub_command == "add-tag":
+        image, tags = split_image_and_tag(image_and_tag_arg)
+        if is_sha256_hash(image):
+          manifest_hash = image
+        else:
+          manifest_hash = tag_to_hash.get(image, None)
+
+        if manifest_hash:
+          manifest_path = hdfs_manifest_dir + "/" + manifest_hash
+          out, err, returncode = hdfs_cat(manifest_path)
+          manifest = json.loads(out)
```

Comment:
unused manifest can be replaced with "_"

Reply:
I think keeping it a as a readable variable is easier to understand even though it is not used